CMakeLists.txt

```diff
@@ -402,18 +393,21 @@ if(WITH_ASAN AND WITH_MSAN)
 endif()
 if(WITH_ASAN)
   set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -g -O1")
-  set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+  list(APPEND cxx_interface_flags -fsanitize=address -fno-omit-frame-pointer)
 endif()
 if(WITH_MSAN)
   set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -g -O1")
-  set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -fsanitize=memory -fno-omit-frame-pointer")
+  list(APPEND cxx_interface_flags -fsanitize=memory -fno-omit-frame-pointer)
 endif()
 if(WITH_UBSAN)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=undefined")
+  list(APPEND cxx_interface_flags -fsanitize=undefined)
 endif()
 
+add_library(cxx_interface INTERFACE)
+target_compile_options(cxx_interface INTERFACE ${cxx_interface_flags})
```

Comment:
out of currriosity, why do you keep the list in `cxx_interface_flags` instead of directly using `target_compile_options`?

Reply:
We actually need to pass these flags to both the compiler and linker, otherwise we get unresolved symbols (e.g. ASAN in [clang:6.0 logfile](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/223148)).

Reply:
In the line below, I pass the same list to [`target_link_libraries()`](https://cmake.org/cmake/help/v3.12/command/target_link_libraries.html?highlight=target_link#id2), which defines linker flags. CMake 3.13 introduced [`target_link_options()`](https://cmake.org/cmake/help/v3.13/command/target_link_options.html) to pass only linker flags, in an effort to make the intent clearer thanks to the name similarity with `target_compile_options()`.

Reply:
Only the `-fsanitize=` no? (c.f.  https://gitlab.icp.uni-stuttgart.de/fweik/mdx/-/blob/master/CMakeLists.txt) Otherwise passing compiler options to the linker doesn't do anything, and is  just confusing. Also you still wouldn't need the list, right?

Reply:
Thanks for the hint! I wasn't sure if `-fsanitize=...` alone was sufficient, but it actually was.

Reply:
Yeah, some of the sanitizers have a runtime component, which needs to be liked...